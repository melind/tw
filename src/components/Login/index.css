

.login {
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 100%;
    width: 100%;
    color: #d5b519;
  }

 .login h1 {
    height: 10%;
  }

.inputLogin {
    
    font-family: 'Roboto Mono', monospace;
    font-size: 100%;
  }

.inputLogin:hover {
    border: 1px solid #db9d01;
  }

.inputLogin:focus {
    border: 1px solid #d6b519;
    box-shadow: 1px 1px #d6b519;
  }


.buttonLogin {
    background-color: #d6b519;
    color: white;
    font-size: 115%;
    margin-left: 30%;
    border: none;
  
  }

.buttonLogin:hover {
    
    background-color: #d6b519;
    color: white;
    font-size: 115%;
    border: none;
  
  }

.buttonLogin:visited{
    
    background-color: #d6b519;
    color: white;
    font-size: 115%;
    border: none;
  
  }

  .login p {
    height: 5%;
    color: #d6b519;
  }
