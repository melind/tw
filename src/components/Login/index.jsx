"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
exports.__esModule = true;
var React = require("react");
var react_1 = require("react");
var react_router_dom_1 = require("react-router-dom");
var antd_1 = require("antd");
require("./index.css");
var Login = function (_a) {
    var pseudo = _a.pseudo, password = _a.password, loggedin = _a.loggedin, subscriber = _a.subscriber, onSubmit = _a.onSubmit, error = _a.error, init = _a.init;
    setTimeout(function () {
        init();
    }, 1000);
    var _b = react_1.useState({ pseudo: pseudo, password: password, loggedin: loggedin, subscriber: subscriber }), formState = _b[0], setFormState = _b[1];
    var handleChange = function (e) {
        var _a;
        var name = e.target.name;
        var value = e.target.value;
        setFormState(__assign(__assign({}, formState), (_a = {}, _a[name] = value, _a))); // name_input : input_value
    };
    var handleSubmit = function (e) {
        e.preventDefault();
        onSubmit(formState);
    };
    if (loggedin)
        return <react_router_dom_1.Redirect to="/home"/>;
    return (<div className="form"> 

        <h1>Connectez-Vous</h1><br />
        
        <form onSubmit={handleSubmit}>
        <label htmlFor="pseudo">Pseudo :  </label><antd_1.Input className="input" id="pseudo" name="pseudo" placeholder="Entrer votre pseudo" onChange={handleChange} value={formState.pseudo} required></antd_1.Input> <br /><br />
        <label htmlFor="password">Mot de passe : </label><antd_1.Input className="input" id="password" name="password" type="password" placeholder="Entrer votre mot de passe" onChange={handleChange} value={formState.password} required></antd_1.Input> <br /><br /><br />
          <antd_1.Button htmlType="submit">Valider</antd_1.Button>
          <p>{error}<br />  
            <a href="/signup">Pas encore inscrit ?</a>
          </p> 
        </form>
        
        </div>);
};
exports["default"] = Login;
