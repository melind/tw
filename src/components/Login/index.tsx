import React, { useState, useEffect} from 'react';
import { Redirect, Link} from 'react-router-dom';
import {Button, Input} from 'antd';
import './index.css';

const Login = ({pseudo, password, loggedin, subscriber, onSubmit, error, init}) => {


 setTimeout(function () {
    init();

  }, 1000);

  console.log("states come from reducer:", pseudo, "abonnÃ© ",subscriber, "erreur ",error);
  const [formState, setFormState] = useState({pseudo, password, loggedin, subscriber});
 
 
  const handleChange = (e) => {
      const name: string = e.target.name;
      const value: string = e.target.value;
      setFormState({...formState, [name]: value}); // name_input : input_value
      
      
      console.log("handleChange :", name, "value: ", value);
  }
 const handleSubmit = (e) => {
        e.preventDefault();
        console.log("props of signup send after submit: ",  formState);
        console.log("value input: ",formState.pseudo);
        
        const {pseudo,  password, subscriber, loggedin} = formState;
        console.log("props update send  by input value:", pseudo, password, subscriber, loggedin);
        onSubmit(formState);
        
        
    }
    if(loggedin) return <Redirect to="/home" />
   
     
     
    return (

        <div className="form"> 

        <h1>Connectez-Vous</h1>< br/>
        
        <form onSubmit={handleSubmit} >
        Pseudo :  <Input className="input" name="pseudo" placeholder="Entrer votre pseudo" onChange={handleChange} value={formState.pseudo} required></Input> < br/>< br/>
        Mot de passe : <Input className="input" name="password" type="password" placeholder="Entrer votre mot de passe" onChange={handleChange} value={formState.password} required></Input> < br/>< br/>< br/>
          <Button  htmlType="submit" >Valider</Button>
          <p>{error}<br/>  
            <Link to="/signup">Pas encore inscrit ?</Link>
          </p> 
        </form>
        
        </div>

    )
    
}

export default Login;