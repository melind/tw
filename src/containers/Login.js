"use strict";
exports.__esModule = true;
var react_redux_1 = require("react-redux");
var Login_1 = require("../components/Login");
var login_reducer_1 = require("../store/reducer/login-reducer");
var mapStateToProps = function (state) { return ({
    pseudo: state.login.pseudo,
    password: state.login.password,
    loggedin: state.login.loggedin,
    subscriber: state.login.subscriber,
    error: state.login.error
}); };
var mapDispatchToProps = function (dispatch) { return ({
    onSubmit: function (formState) {
        dispatch(login_reducer_1.login(formState));
    },
    init: function () {
        dispatch(login_reducer_1.init());
    }
}); };
exports["default"] = react_redux_1.connect(mapStateToProps, mapDispatchToProps)(Login_1["default"]);
