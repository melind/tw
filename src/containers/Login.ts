import { connect } from 'react-redux';
import Login from '../components/Login';
import { login } from '../store/reducer/login-reducer';

const mapStateToProps = /*(state: {//state detrermine by reducer
    pseudo: string,
    password: string,
    loggedin: boolean,
    subscriber: boolean
 }) => ({ //props compnonent
    pseudo: state.login.pseudo,
    password: state.login.password,
    loggedin: state.login.loggedin,
    subscriber: state.login.subscriber
 });*/
 (state, props) => ({
   pseudo: state.login.pseudo,
    password: state.login.password,
    loggedin: state.login.loggedin,
    subscriber: state.login.subscriber
});

const mapDispatchToProps = (dispatch, props) => ({
   // fetchRoom: () => {
   //     dispatch(fetchRoom(props.match.params.id))
   // }
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);