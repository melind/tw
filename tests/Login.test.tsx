import React from 'react';
import { shallow, mount, render} from 'enzyme';
import renderer from 'react-test-renderer';
import Login from '../src/components/Login';

/** test to check that we don't broke our code in the evolution */

/* mock = simulating  */



/*------ mock function --------*/

const onSubmitMock = jest.fn();
const initMock = jest.fn();


/*------ mock event -----------*/

//the code testing uses the event so we have to mock it 
// mock method preventDefault() of event
const fakeEvent = { preventDefault: () => "" };

const props = {
  pseudo : "", 
  password : "", 
  loggedin: false, 
  subscriber:true, 
  onSubmit: () => onSubmitMock(), 
  error:"", 
  init: () => initMock()
};

const formState = {pseudo: "", password: "", loggedin: false, subscriber: true};

/* snapshot */

/* snapshot tool let us to follow the evlution of our app,
 it compare the dom of our component save in file snap and the actual dom*/

describe('Login', () => {
it('Login should render correctly', () => {
  const tree = renderer
    .create(<Login {...props}/>)
    .toJSON(); 
  expect(tree).toMatchSnapshot();
});

/* unit test*/

 it('renders Login', () => {
     const wrapper = shallow(<Login  {...props}/>);
     expect(wrapper.length).toEqual(1);   
  }); 

it('should render 2 input ', () => {
     const wrapper = shallow(<Login {...props}/>);
      expect(wrapper.find("label").length).toBe(2);   
  }); 
  
  it('Should call "onSubmit" inside the handleSubmit (the props) when submit form', () => {
    const wrapper = shallow(<Login {...props}/>);
    wrapper.find('form').simulate('submit', fakeEvent);
    expect(onSubmitMock).toHaveBeenCalled();
  }); 

it('Should be equal to value before change ' , () => {
    const wrapper = shallow(<Login {...props}/>);
    expect(wrapper.find('.input').at(0).prop('value')).toEqual('');
    
  }); 


});
