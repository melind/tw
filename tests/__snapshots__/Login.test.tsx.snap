// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Login should render correctly 1`] = `
<div
  className="form"
>
  <h1>
    Connectez-Vous
  </h1>
  <br />
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="pseudo"
    >
      Pseudo :  
    </label>
    <input
      className="ant-input input"
      id="pseudo"
      name="pseudo"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Entrer votre pseudo"
      required={true}
      type="text"
      value=""
    />
     
    <br />
    <br />
    <label
      htmlFor="password"
    >
      Mot de passe : 
    </label>
    <input
      className="ant-input input"
      id="password"
      name="password"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Entrer votre mot de passe"
      required={true}
      type="password"
      value=""
    />
     
    <br />
    <br />
    <br />
    <button
      className="ant-btn"
      onClick={[Function]}
      type="submit"
    >
      <span>
        Valider
      </span>
    </button>
    <p>
      
      <br />
      <a
        href="/signup"
      >
        Pas encore inscrit ?
      </a>
    </p>
  </form>
</div>
`;
